src/pip/_internal/commands/completion.py

```diff
@@ -34,10 +34,8 @@
     """,
     'fish': """
         function __fish_complete_pip
-            set -lx COMP_WORDS (commandline -o) ""
-            set -lx COMP_CWORD ( \\
-                math (contains -i -- (commandline -t) $COMP_WORDS)-1 \\
-            )
+            set -lx COMP_WORDS (commandline -opc) (commandline -ct)
+            set -lx COMP_CWORD (math (count $COMP_WORDS) - 1)
             set -lx PIP_AUTO_COMPLETE 1
             string split \\  -- (eval $COMP_WORDS[1])
```

Comment:
This could be changed to remove the `eval`, though that would bump the minimum required fish version to fish 3.0.0 (released December 28, 2018).

Reply:
thanks, I have changed this to only run eval for version 2.*